### PR TITLE
perf(zk): merge parallelized parts

### PR DESCRIPTION
**Pull request by dongchangYoo**
In this PR, the calls to base::Parallelize() that were happening within `UpdateValuesByCustomGates`, `UpdateValuesByPermutation`, and `UpdateValuesByLookups` have been modified to occur just once from the caller.
This PR does not result in any current performance changes, but it has improved code readability and made it easier to optimize parallelization in the future.

### CODE REVIEW

**Review by chokobole — APPROVED** [review]
LGTM

**Review by ashjeong — COMMENTED** [review]
https://github.com/kroma-network/tachyon/pull/297/commits/ed39cd091d881e9b21fc070ad34972d6c6ac425f
“merge _**seperated parallized**_ parts in CircuitPolynomialBuilder” to “merge separated parallelized parts in CircuitPolynomialBuilder”

https://github.com/kroma-network/tachyon/pull/297/commits/eac09ac89bfdb48a63e01eae802956a47a60fa37
“move shared values out of _**foor**_ loop” to “move shared values out of for loop”

note: I added the pr scope as "(zk)"

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

File: tachyon/zk/plonk/vanishing/circuit_polynomial_builder.h [hunk 1]
```diff
@@ -130,132 +135,127 @@ class CircuitPolynomialBuilder {
...
+        chunk[j] *= *y_;
+        chunk[j] +=
+            (product_coset[idx].Square() - product_coset[idx]) * l_last_[idx];
 
-          // clang-format off
+        // clang-format off
           // A * (B - C) = 0 where
           //  - A = 1 - (l_last(X) + l_blind(X))
           //  - B = z(wX) * (a'(X) + β) * (s'(X) + γ)
           //  - C = z(X) * (θᵐ⁻¹ a₀(X) + ... + aₘ₋₁(X) + β) * (θᵐ⁻¹ s₀(X) + ... + sₘ₋₁(X) + γ)
```
Comment (Insun35) [inline tachyon/zk/plonk/vanishing/circuit_polynomial_builder.h:173]: Please move the indentation of this comment back as well.

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

**Pull request merged**